Prototope/Screen.swift

```diff
@@ -0,0 +1,224 @@
+		let window = UIApplication.sharedApplication().delegate?.window
+		window!!.addSubview(overlayView!)
+	}
+	
+	
+	private static func removeTouchDots() {
+		overlayView?.removeFromSuperview()
+	}
+}
+
+
+/** Internal view subclass to act as a passthrough and to show touch dots. */
+class TouchDotOverlayView: UIView {
+	
+	let gestureRecognizer = TouchDotGestureRecognizer()
+	var dotViews = [Int: UIImageView]()
+	
+	override init(frame: CGRect) {
+		super.init(frame: frame)
+		self.autoresizingMask = .FlexibleHeight | .FlexibleWidth
+		self.userInteractionEnabled = true
+		self.backgroundColor = UIColor.clearColor()
+		self.opaque = false
+		
+		self.gestureRecognizer.touchDelegate = self
+	}
+
+	required init(coder aDecoder: NSCoder) {
+	    fatalError("init(coder:) has not been implemented")
+	}
+}
+
+
+// UIView hierarchy overrides
+extension TouchDotOverlayView {
+	
+	override func didMoveToSuperview() {
+		self.gestureRecognizer.view?.removeGestureRecognizer(self.gestureRecognizer)
+		self.superview?.addGestureRecognizer(self.gestureRecognizer)
+	}
+	override func hitTest(point: CGPoint, withEvent event: UIEvent?) -> UIView? {
+		return nil
+	}
+	
+	override func pointInside(point: CGPoint, withEvent event: UIEvent?) -> Bool {
+		return false
+	}
+}
+
+
+// Touch dot management
+extension TouchDotOverlayView {
+	func updateTouch(touch: UITouch) {
+		let location = touch.locationInView(self)
+		let key = touch.hash
+		
+		var view = self.dotViews[key]
+		if view == nil {
+			let image = UIImage(named: "finger", inBundle: NSBundle(forClass: self.dynamicType), compatibleWithTraitCollection: nil)
+			view = UIImageView(image: image)
+			view?.sizeToFit()
+			self.addSubview(view!)
+			self.dotViews[key] = view
+		}
+		
+		view?.center = location
+	}
+	
+	
+	func removeViewForTouch(touch: UITouch) {
+		let view = self.dotViews.removeValueForKey(touch.hash)
+		view?.removeFromSuperview()
+	}
+}
+
+
+// Touch handling from the gesture recognizer
+extension TouchDotOverlayView: TouchDotGestureRecognizerDelegate {
+	func touchesBegan(touches: Set<UITouch>) {
+		// ensure we are topmost
+		self.superview?.bringSubviewToFront(self)
+		for touch in touches {
+			self.updateTouch(touch)
+		}
+	}
+	
+	
+	func touchesMoved(touches: Set<UITouch>) {
+		for touch in touches {
+			self.updateTouch(touch)
+		}
+	}
+	
+	
+	func touchesEnded(touches: Set<UITouch>) {
+		for touch in touches {
+			self.removeViewForTouch(touch)
+		}
+	}
+	
+}
+
+
+/** Internal gesture recognizer class to handle where touches are and how they interact with other GR in the system. */
+class TouchDotGestureRecognizer: UIGestureRecognizer, UIGestureRecognizerDelegate {
+	
+	var activeTouches = Set<UITouch>()
+	weak var touchDelegate: TouchDotGestureRecognizerDelegate?
+
+	init() {
+		
+		// TODO(jb): I'm not sure how to hack around this, I don't intend on messaging the target, but I've got to initialize super...
+		super.init(target: NSObject(), action: "description")
```

Comment:
Only because I was basically re-writing the ObjC source I linked to and that’s what it did (and didn’t feel like spending too much time futzing with it). But I can go ahead and write it the better way you suggest if you’d like.